lib/graphql-federation.factory.ts

```diff
@@ -0,0 +1,100 @@
+import { Injectable } from '@nestjs/common';
+import { gql } from 'apollo-server-core';
+import { loadPackage } from '@nestjs/common/utils/load-package.util';
+import { extend } from './utils';
+import { isEmpty, forEach } from 'lodash';
+import {
+  ScalarsExplorerService,
+  DelegatesExplorerService,
+  ResolversExplorerService,
+} from './services';
+import { mergeSchemas } from 'graphql-tools';
+import { GqlModuleOptions } from './interfaces';
+import { GraphQLSchemaBuilder } from './graphql-schema-builder';
+import { GraphQLFactory } from './graphql.factory';
+import { GraphQLSchema, GraphQLSchemaConfig } from 'graphql';
+import { GraphQLObjectType } from 'graphql';
+
+@Injectable()
+export class GraphQLFederationFactory {
+  constructor(
+    private readonly resolversExplorerService: ResolversExplorerService,
+    private readonly delegatesExplorerService: DelegatesExplorerService,
+    private readonly scalarsExplorerService: ScalarsExplorerService,
+    private readonly gqlSchemaBuilder: GraphQLSchemaBuilder,
+    private readonly graphqlFactory: GraphQLFactory,
+  ) {}
+
+  async mergeOptions(options: GqlModuleOptions = {}): Promise<GqlModuleOptions> {
+    const transformSchema = async s => (options.transformSchema ? options.transformSchema(s) : s);
+
+    let schema: GraphQLSchema;
+    if (options.autoSchemaFile) {
+      // Enable support when Directive support in type-graphql goes stable
+      throw new Error('Code-first not supported yet');
+      schema = await this.generateSchema(options);
+    } else if (!isEmpty(options.typeDefs)) {
```

Comment:
I think it's supposed to be:

```ts
} else if (!isEmpty(options.schema)) {
```

Reply:
Nope, it follows the design from the original `graphql.factory.ts`. If you don't pass an array of typeDefs that will be used to construct a schema, then you are required to provide your own schema.

Reply:
so shouldn't that be:
```typescript
else if (isEmpty(options.typeDefs)) {
```
as it is in https://github.com/nestjs/graphql/blob/master/lib/graphql.factory.ts#L77 ?

Now if you try to instantiate passing typePaths in options, it will throw an error because options.schema is undefined, if you pass an existing schema with options.schema, it will throw because it try to `buildSchemaFromTypeDefs` where typeDefs is `null`

Reply:
😅 you are absolutely correct. Thank you very much!

I forgot to disable the unittest filter when I last pushed this branch, it was still set only running type-graphql tests.